status/views.py

```diff
@@ -1,97 +1,109 @@
-                data[2]['details'] = '\n'.join(reasons)
+
+class StatusView(APIView):
+    permission_classes = (IsAdminUser,)
+
+    def get(self, request):
+        connection_info = []
+
+        redis_status, redis_details = self.test_redis_connection()
+        connection_info.append({"name": "Redis", "status": redis_status, "details": redis_details})
+
+        postgres_status, postgres_details = self.test_postgresql_connection()
+        connection_info.append({"name": "PostgreSQL", "status": postgres_status, "details": postgres_details})
+
+        andromeda_status, andromeda_details = self.get_andromeda_status()
+        connection_info.append({"name": "Andomeda", "status": andromeda_status, "details": andromeda_details})
+
+        if settings.SEND_MAIL:
+            mailusv_status, mailusv_details = self.test_mailusv_connection()
+            connection_info.append({"name": "Mail Daemon", "status": mailusv_status, "details": mailusv_details})
+
+        return FormattedResponse(connection_info)
+
+    def test_redis_connection(self):
+        status, details = 'unknown', ''
+
+        try:
+            rs_details = settings.CONFIG['REDIS']
+            rs = redis.Redis(host=rs_details['HOST'], port=rs_details['PORT'])
+
+            if rs.ping() == True:
+                status = 'online'
             else:
-                data[2]['status'] = 'online'
-        else:
-            data[2]['status'] = 'unknown'
-            data[2]['details'] = 'Error connecting to /status'
-    except Exception as e:
-        data[2]['status'] = 'offline'
-        data[2]['details'] = str(e)
-
-    if settings.SEND_MAIL:
-        data.append(
-        {
-            "name": "Mail Daemon",
-            "status": "unknown",
-            "details": ''
-        })
+                status = 'offline'
+                details = "PING failed"
+        except redis.exceptions.RedisError as e:
+            status = 'offline'
+            details = "PING failed:\n" + str(e)
+
+        return status, details
+
+    def test_postgresql_connection(self):
+        status, details = 'unknown', ''
+
         try:
-            response = requests.get(settings.MAIL_SOCK_URL)
-            if response.status_code != 200:
-                data[3]['status'] = 'unknown'
-                data[3]['details'] = 'Unknown error whilst connecting to /'
+            with connection.cursor() as cursor:
+                cursor.execute("SELECT 1");
+            status = 'online'
+        except psycopg2.OperationalError as e:
+            status = 'offline'
+            details = 'Could not execute query `SELECT 1\':\n' + str(e)
+        
+        return status, details
+
+    def get_andromeda_status(self):
+        status, details = 'unknown', ''
+        try:
+            response = requests.get(f"{settings.CHALLENGE_SERVER_URL}/status",
+                                headers={"Authorization": settings.CHALLENGE_SERVER_API_KEY})
+            if response.status_code == 200:
```

Comment:
Minor thing, but to fit the rest of the codebase's convention better, use the HTTP_200_OK constant from rest framework